libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -517,13 +518,12 @@ uint64_t VoteManager::determineRoundFromPeriodAndVotes(uint64_t period, size_t t
         LOG(log_nf_) << "Round determined for period " << period << ". Found " << count << " votes for voted value "
                      << voted_value.first << " for round " << round_rit->first << ", step " << step_rit->first;
 
-        next_votes_manager_->updateNextVotes(votes, two_t_plus_one);
-        return round_rit->first + 1;
+        return {std::make_pair(round_rit->first + 1, std::move(votes))};
```

Comment:
I see, you wanted to move this updating of next votes out of this function I guess so thats why you return votes?

Reply:
yes exactly